CReLU.lua

```diff
@@ -0,0 +1,51 @@
+local CReLU, parent = torch.class('nn.CReLU', 'nn.Sequential')
+
+-- Implements the CReLU activation function as described by
+-- W. Shang et al. in "Understanding and Improving Convolutional Neural Networks
+-- via Concatenated Rectified Linear Units"
+-- Note: Only supports batched input of the form [B,C,H,W]
+function CReLU:__init(inplace)
+   parent.__init(self)
+   local concatTable = nn.ConcatTable()
+   concatTable:add(nn.Identity())
+   concatTable:add(nn.MulConstant(-1))
+   self.inplace = inplace or false
+   self.batched = true
+   self:add(concatTable)
+   self:add(nn.JoinTable(2))
+   self:add(nn.ReLU(self.inplace))
+end
+
+
+function CReLU:updateOutput(input)
+   self.batched = true
+   if input:nDimension() == 3 then
+      self.batched = false
+   end
+
+   if self.batched then
+      parent.updateOutput(self, input)
+   else
+      parent.updateOutput(self, input:view(1, input:size(1), input:size(2), input:size(3)))
```

Comment:
you can fix CReLU to not depend on 3D and 4D inputs.
Considering that it is CReLU and not SpatialCReLU, please do the following:

- remove the self.batched checks...
Do:
```
function CReLU:updateOutput(input)
   parent.updateOutput(self, input:view(1, -1))
   self.output:resizeAs(input)
   return self.output
end
```
make similar changes for updateGradInput